tests/test_kwargs_handlers.py

```diff
@@ -96,6 +96,46 @@ def test_autocast_kwargs(self):
             # We should be back in fp16
             assert g_float16.dtype == torch.float16
 
+    def test_profile_kwargs(self):
+        # Arrange
+        schedule_options = [
+            dict(wait=1, warmup=1, active=2, repeat=1),
+            dict(wait=2, warmup=2, active=2, repeat=2),
+            dict(wait=0, warmup=1, active=3, repeat=3, skip_first=1),
+            dict(wait=3, warmup=2, active=1, repeat=1, skip_first=2),
+            dict(wait=1, warmup=0, active=1, repeat=5),
+        ]
+
+        total_steps = 100
+
+        for option in schedule_options:
+            count = 0
+            steps_per_cycle = option["wait"] + option["warmup"] + option["active"]
+            effective_steps = max(0, total_steps - option.get("skip_first", 0))
+            cycles = effective_steps // steps_per_cycle
+            if option["repeat"] > 0:
+                expected_count = min(cycles, option["repeat"])
+            else:
+                expected_count = cycles
+
+            def on_trace_ready(prof):
+                nonlocal count
+                count += 1
+                print(prof.key_averages().table(sort_by="cpu_memory_usage", row_limit=-1))
```

Comment:
I wonder: Instead of printing, could we, for instance, just add this to a list, then do a basic sanity check on the output, for example what columns we should expect?